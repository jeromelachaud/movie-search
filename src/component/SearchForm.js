import React, { Component } from 'react';
import SearchResults from './SearchResults';
import _bind from 'lodash/function/bind';
import './SearchForm.css';
import axios from 'axios';
import { apiKey } from '../constants/api-constants';

export default class SearchForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      results: [],
      query: ''
    };

    this.onSubmit = _bind(this.onSubmit, this);
    this.onChangeQuery = _bind(this.onChangeQuery, this);
  }

  onChangeQuery(event) {
    this.setState({query: event.target.value});
  }

  render() {
    return (
      <div>
        <form
          onSubmit={this.onSubmit}>
          <div
            className="search-form">
            <label
              className="search-form-label"
              htmlFor="MovieSearch">
              Search in TMDb
            </label>
            <input
              aria-label="Search for a movie in The Movie Database"
              className="search-form-input-field"
              id="searchFormInPutField"
              onChange={this.onChangeQuery}
              placeholder="Search for a movie"
              required
              size="80"
              type="search">
            </input>
            <button
              className="search-form-button"
              id="searchFormButton"
              type="submit">
              Search!
            </button>
          </div>
        </form>
        <SearchResults
          results={this.state.results}>
        </SearchResults>
      </div>
    );
  }

  onSubmit(e) {
    e.preventDefault();
    const url = `https://api.themoviedb.org/3/search/movie?&api_key=${apiKey}&language=en-US&query=${encodeURI(this.state.query)}&page=1&include_adult=false`;
    axios.get(url)
      .then(response => {
        const results = {
          results: response.data.results
        };
        this.setState(results);
      })
      .catch((error) => {
        throw error;
        // console.log(error);
      });
  }
}
