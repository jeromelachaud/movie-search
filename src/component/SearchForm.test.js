import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  const searchFormComponent = shallow(<SearchForm />);

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchForm />, div);
  });

  it('should render as expected', () => {
    expect(toJson(searchFormComponent))
      .toMatchSnapshot();
  });

  it('should add the query to the state', () => {
    const input = searchFormComponent.find('#searchFormInPutField');
    input.simulate('change', { target: { value: 'Hello' } });
    expect(searchFormComponent.state().query)
      .toBe('Hello');
  });
});
