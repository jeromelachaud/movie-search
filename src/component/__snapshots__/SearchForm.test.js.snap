// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render as expected 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="search-form"
    >
      <label
        className="search-form-label"
        htmlFor="MovieSearch"
      >
        Search in TMDb
      </label>
      <input
        aria-label="Search for a movie in The Movie Database"
        className="search-form-input-field"
        id="searchFormInPutField"
        onChange={[Function]}
        placeholder="Search for a movie"
        required={true}
        size="80"
        type="search"
      />
      <button
        className="search-form-button"
        id="searchFormButton"
        type="submit"
      >
        Search!
      </button>
    </div>
  </form>
  <SearchResults
    results={Array []}
  />
</div>
`;
